### ISSUE TITLE
Istio generating different envoy config for identical services

### ISSUE THREAD

**Issue by clayvan** [istio/istio#42467]
### Bug Description

So I’m trying to understand how Istio Pilot is generating the envoy configuration for two services differently even with identical istio config

From my reading, _something_ is reading the incoming traffic and determining what sort of applicationProtocols and other config it should set on the listeners

However I have two services. `A`  and  `B` , the only difference being that the destination for service `B` is actually a traefik service to do some middleware. All pods have istio sidecars enabled (including traefik). I’m using istio ingress gateway in front. Both service `A` and `traefik` are fronted by HTTP k8s services. Service `B` however is an HTTPS service (This might be the difference that matters?).

- When I send a request to `A` through the istio gateway, the outcome is expected (envoyfilter with Lua scripts runs).
- When I send a request to `B` through the istio gateway, the outcome is that my envoy filter is not getting triggered.
- When I send a request to `B` bypassing istio gateway via k8s service, the outcome is expected (envoyfilter with Lua scripts run)

When I drill into the Envoy listener config, I see the two services configured applicationProtocols differently,

`A` has the following protocols
```
                    "applicationProtocols": [
                        "istio",
                        "istio-peer-exchange",
                        "istio-http/1.0",
                        "istio-http/1.1",
                        "istio-h2"
                    ]
```
`B` has these
```
                    "applicationProtocols": [
                        "istio-http/1.0",
                        "istio-http/1.1",
                        "istio-h2"
                    ]
```

So my theory is that even though the virtual services and envoyfilters are configured identical for these 2 services, the `A` traffic is somehow seen as `istio` or `istio-peer-exchange` when being sent through the istio ingress gateway, but my `B` traffic is not.

What am I missing here? Ideally I don't have to add an EnvoyFilter for patching the listener config on `B`, nor do I know if that's even possible to achieve my goal.

VirtualService:
```
kind: VirtualService
apiVersion: networking.istio.io/v1beta1
metadata:
  name: example
  namespace: default
spec:
  hosts:
    - myhost
  gateways:
    - mygateway
  http:
    - match:
        - uri:
            prefix: /a
      route:
        - destination:
            host: a.default.svc.cluster.local
    - match:
        - uri:
            prefix: /b
      route:
        - destination:
            host: traefik.default.svc.cluster.local
```
EnvoyFilter
```
apiVersion: networking.istio.io/v1alpha3
kind: EnvoyFilter
metadata:
  name: myappfilters
  namespace: default
spec:
  configPatches:
    - applyTo: HTTP_FILTER
      match:
        context: SIDECAR_INBOUND
        listener:
          filterChain:
            filter:
              name: "envoy.filters.network.http_connection_manager"
      patch:
        operation: INSERT_BEFORE
        value:
          name: envoy.lua
          typed_config:
            "@type": "type.googleapis.com/envoy.extensions.filters.http.lua.v3.Lua"
            inlineCode: |
              function envoy_on_request(request_handle)
                request_handle:logWarn("hello world")

```

### Version

```prose
$ istioctl version
client version: 1.14.3
control plane version: 1.13.7
data plane version: 1.13.7 (81 proxies)

$ kubectl version --short
Client Version: v1.24.1
Kustomize Version: v4.5.4
Server Version: v1.21.14-eks-fb459a0
```

**Comment by howardjohn** [comment]
It should be based on thr https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/. Are you sure the services are identical?

**Comment by howardjohn** [comment]
Oh you said one is HTTPs, so yes, that is the difference

**Comment by clayvan** [comment]
Oh wow, @howardjohn thank you for the reference. The service that is HTTPs is actually behind traefik which is HTTP, so I wouldn't have expected istio to care about it.

However, that document did remind me to go check on the naming of service Ports. Turns out the traefik helm chart names it `web`, when I changed this to `http` sure enough the envoy listener config matched service A.

Thank you very much, I would not have even thought about that. Closing this out.
